src/Solana/useSolana.tsx

```diff
@@ -0,0 +1,391 @@
+
+        if (opts.makerName && hemProgram) {
+          const maker = makerKey(hntDaoKey, opts.makerName)[0]
+          const makerAcc = await hemProgram.account.makerV0.fetch(
+            maker.toString()
+          )
+          searchParams.collection = makerAcc.collection.toString()
+        }
+      }
+
+      return searchAssets(METAPLEX_URL, searchParams)
+    },
+    [hemProgram, vars?.hnt.mint, wallet]
+  )
+
+  const estimateMetaTxnFees = useCallback(
+    async (buff: Buffer, { maker }: { maker: PublicKey }) => {
+      if (!wallet || !vars?.dc.mint) return
+
+      const walletDC = await getAssociatedTokenAddress(
+        new PublicKey(vars.dc.mint),
+        wallet
+      )
+
+      const makerDC = await getAssociatedTokenAddress(
+        new PublicKey(vars.dc.mint),
+        maker
+      )
+
+      const [makerAccount, makerDcAccount, ownerAccount, ownerDcAccount] =
+        await fetchSimulatedTxn({
+          apiUrl: clusterApiUrl(cluster),
+          accountAddresses: [
+            maker.toString(),
+            makerDC.toString(),
+            wallet.toString(),
+            walletDC.toString(),
+          ],
+          txnBuff: buff,
+        })
+
+      const fees = await estimateFees({
+        connection,
+        owner: {
+          key: wallet,
+          account: ownerAccount,
+          dcAccount: ownerDcAccount,
+        },
+        maker: { key: maker, account: makerAccount, dcAccount: makerDcAccount },
+        dcMint: new PublicKey(vars.dc.mint),
+      })
+      return fees
+    },
+    [cluster, connection, vars, wallet]
+  )
+
+  return {
+    connection,
+    createTransferCompressedCollectableTxn,
+    estimateMetaTxnFees,
+    getDcBalance,
+    getHntBalance,
+    getBalances,
+    getHotspots,
+    getOraclePriceFromSolana,
+    getSolBalance,
+    status: {
+      inProgress,
+      isHelium,
+      isSolana,
+    },
+    submitSolana,
+    submitAllSolana,
+    vars,
+  }
+}
+
+export const getAccountFees = async ({
+  dcAccount,
+  account,
+  connection,
+  key,
+  dcMint,
+}: {
+  key: PublicKey
+  dcAccount?: Account
+  account?: Account
+  connection: Connection
+  dcMint: PublicKey
+}) => {
+  const lamportsBefore = await connection.getBalance(key)
+  const dcBefore = Number(
+    await getBalance({ pubKey: key, mint: dcMint, connection })
+  )
+
+  let lamportsAfter = 0
+  let dcAfter = 0
+
+  if (account) {
+    lamportsAfter = BigNumber(account.lamports.toString()).toNumber()
+  }
+
+  const lamportFee = lamportsBefore - lamportsAfter
+  let dcFee = 0
+
+  if (dcAccount) {
+    const tokenAccount = AccountLayout.decode(
+      Buffer.from(dcAccount.data[0], dcAccount.data[1] as BufferEncoding)
+    )
+
+    const dcBalance = BigNumber(tokenAccount.amount.toString())
+    dcAfter = dcBalance.toNumber()
+    dcFee = dcBefore - dcAfter
+    // TODO: will dc show as negative if they don't have enough?
+  } else if (lamportFee) {
+    // TODO: they have no dc and they are the payer, now what?
```

Comment:
You'd need to burn hnt for DC.

Example here:

https://github.com/helium/helium-program-library/blob/master/tests/data-credits.ts#L202